doc/guides/coding_guidelines/index.rst

```diff
@@ -0,0 +1,778 @@
+.. _coding_guidelines_misra:
+
+Coding Guidelines
+#################
+
+The project TSC and the Safety Committee of the project to implement a staged
+and incremental approach for complying with a set of coding rules (AKA Coding
+Guidelines) to improve quality and consistency of the code base. Below are the
+agreed upon stages and the approximate timelines:
+
+Stage I
+  Coding guideline rules are available to be followed and referenced,
+  but not enforced. Rules are not yet enforced in CI and pull-requests cannot be
+  blocked by reviewers/approvers due to violations.
+
+Stage II
+  Begin enforcement on a limited scope of the code base. Initially this would be
+  safety certification scope. For rules easily applied across codebase, we
+  should not limit compliance to initial scope. This step requires tooling and
+  CI setup.
+  The enforcement for safety certification scope to begin during the 2.4
+  development cycle and end with the Zephyr LTS2 (2.6) to achieve and LTS2 that
+  is complaint for certification.
+
+Stage III
+  Revisit the coding guideline and based on experience from previous stages,
+  refine/iterate on selected rules. This stage is to statr approximataly after LTS2
+
+Stage IV
+   Expand enforcement to wider codebase where exceptions can be applied on some
```

Comment:
This reads a bit easier if you say '...to the wider codebase.  Exceptions may be granted on some...'

Also, you could copy the MISRA C wording about exemptions pretty much directly. here.